platforms/android/demo/src/main/java/com/mapzen/tangram/android/MainActivity.java

```diff
@@ -88,6 +191,42 @@ public void onLowMemory() {
         view.onLowMemory();
     }
 
+    @Override public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        int selectedIndex = 7; // custom
+        String sceneText = autoCompleteView.getText().toString();
+        for (int i = 0; i < scenes.length; i++) {
+            if (sceneText.equals(scenes[i])) {
+                selectedIndex = i;
+                break;
+            }
+        }
+        spinner.setSelection(selectedIndex);
+
+        autoCompleteView.clearFocus();
+        InputMethodManager in = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        in.hideSoftInputFromWindow(autoCompleteView.getWindowToken(), 0);
+
+        map.loadSceneFile(autoCompleteView.getText().toString(),sceneUpdates);
```

Comment:
Loading a scene file synchronously from a network location on the UI thread is ... _no bueno_. Android could very well kill our app or produce the famous "Application Not Responding" message if this takes too long. However, there does seem to be a problem with calling `MapView$getMapAsync` more than once per `MapView` ... so either we live with an unresponsive UI for now or we fix `getMapAsync`.